File: Bookito/app/src/main/java/com/nunez/bookito/customViews/BookTextWatcher.java
package com.nunez.bookito.customViews;

import android.text.Editable;
import android.text.TextWatcher;

import com.nunez.bookito.searchBooks.SearchBooksContract;

/**
 * Created by paulnunez on 3/21/17.
 */

public class BookTextWatcher implements TextWatcher {

  private SearchBooksContract.View view;

  public BookTextWatcher(SearchBooksContract.View view) {
    this.view = view;

  }

  @Override
  public void beforeTextChanged(CharSequence s, int start, int count, int after) {

  }

  @Override
  public void onTextChanged(CharSequence s, int start, int before, int count) {
   }

  @Override
  public void afterTextChanged(Editable s) {
    view.onSearchTextChange(s.toString());
  }
}

File: Bookito/app/src/main/java/com/nunez/bookito/customViews/BookTextWatcher.kt
package com.nunez.bookito.customViews

import android.text.Editable
import android.text.TextWatcher

/**
 * Created by paulnunez on 3/21/17.
 */

class BookTextWatcher(val listener: (String) -> Unit) : TextWatcher {

    override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) {}

    override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {}

    override fun afterTextChanged(s: Editable) {
        listener.invoke(s.toString())
    }

    fun onTextChangedListener(listener: (String) -> Unit) {}

}
